src/core/unit.c

```diff
@@ -5450,7 +5450,13 @@ int unit_prepare_exec(Unit *u) {
 
         /* Prepares everything so that we can fork of a process for this unit */
 
-        (void) unit_realize_cgroup(u);
+        r = unit_realize_cgroup(u);
+        if (r < 0 && IN_SET(u->type, UNIT_MOUNT, UNIT_SWAP)) {
```

Comment:
This chunk should not be part of this commit.

Reply:
I've changed this a bit based on input from @Werkov and split it into separate commit.